### ISSUE TITLE
[feed]:Improve ML model performance and prediction accuracy

### ISSUE THREAD

**Issue by arfahai** [Arnav-Singh-5080/IPL-Win-Probability-Prediction#546]
## Problem

The current cricket win prediction model provides a good baseline performance, but there is room to improve prediction accuracy and generalization.

Currently the model uses features such as:

- batting team
- bowling team
- venue/city
- target
- runs left
- balls left
- wickets remaining
- current run rate (CRR)
- required run rate (RRR)

The existing pipeline mainly relies on Logistic Regression / Random Forest / XGBoost with basic feature engineering.

## Proposed Improvements

Explore ML improvements to increase prediction quality:

### Feature Engineering
- Add recent team form features
- Add batting/bowling strength indicators
- Add venue-based historical performance
- Add toss impact features
- Add pressure/match situation features
- Improve handling of wickets remaining
- Add innings phase features (powerplay, middle overs, death overs)

### Model Experimentation
Compare different approaches:

- Hyperparameter tuning for existing models
- RandomizedSearchCV / GridSearchCV
- XGBoost parameter optimization
- Try additional classifiers
- Ensemble approaches

### Evaluation Improvements

Improve evaluation pipeline:

- Compare models using multiple metrics:
  - Accuracy
  - Precision
  - Recall
  - F1-score
  - ROC-AUC
  - Calibration

- Add better cross-validation strategy

## Expected Outcome

A more reliable win probability prediction model with improved accuracy and better generalization on unseen matches.

## Current Baseline

Current model performance:

- Accuracy: ~81%
- CV Accuracy: ~81%
- F1 Score: ~82%

Goal:
Improve prediction reliability while avoiding overfitting.

<img width="1472" height="740" alt="Image" src="https://github.com/user-attachments/assets/1981712a-fabc-4f77-afa1-0fd432e4a7b8" />

**Comment by Sachin-Rathore-1234** [comment]
Hi @arfahai, I'd like to work on this issue.
I'll start with the feature engineering and evaluation improvements since those are likely to give the most reliable accuracy gains:

Recent team form, batting/bowling strength ratings, venue-based historical performance, toss impact, and innings-phase (powerplay/middle/death) features
A weighted "wickets remaining" metric (resource-based, not just raw count)
Switch to GroupKFold cross-validation at the match level to rule out leakage from ball-by-ball rows of the same match appearing in both train and test
Hyperparameter tuning for XGBoost (Optuna/RandomizedSearchCV) and a simple ensemble (LR + RF + XGB)
Add precision, recall, ROC-AUC, and calibration (reliability diagram) to the evaluation pipeline alongside accuracy/F1

Could you assign this to me? I'll open a PR with the updated pipeline and a before/after metrics comparison.
